crates/circuit/src/operations.rs

```diff
@@ -2043,6 +2057,155 @@ impl Operation for StandardGate {
     fn standard_gate(&self) -> Option<StandardGate> {
         Some(*self)
     }
+    fn inverse(&self, params: &[Param]) -> Option<(StandardGate, Vec<Param>)> {
```

Comment:
Everywhere else we work with parameters in Rust we tend to use `SmallVec<[Param; 3]>` to keep things on the stack unless there are more than 3 elements. It might be worth changing this and updating the `vec!` macro usage to `smallvec!`.